modules/tm/t_reply.c

```diff
@@ -268,8 +268,8 @@ inline static int update_totag_set(struct cell *t, struct sip_msg *ok)
 
 	if (!s || !n) {
 		LM_ERR("no more share memory \n");
-		if (n) shm_free(n);
-		if (s) shm_free(s);
+		if (n) shm_free_bulk(n);
+		if (s) shm_free_bulk(s);
```

Comment:
Hi, Peter!

This is not OK.  The purpose of `shm_free_bulk()` is to be called within a `shm_lock() / shm_unlock()` block.  It is incorrect to call this function in any other context, as the code will break if you use either of the `F_MALLOC` or `Q_MALLOC` allocators.

Regarding that specific `if` block: it is correct as it is -- no need for any change there.

Reply:
Ah ok, I can see your point. Yes this is a mistake.